components/scream/tests/rrtmgp/rrtmgp_stand_alone.cpp

```diff
@@ -1,32 +1,171 @@
+            inputfile, ncol, p_lay, t_lay, p_lev, t_lev, gas_concs, col_dry,
+            sfc_alb_dir, sfc_alb_dif, mu0,
+            lwp, iwp, rel, rei
+        );
+
+        // Setup flux outputs; In a real model run, the fluxes would be
+        // input/outputs into the driver (persisting between calls), and
+        // we would just have to setup the pointers to them in the
+        // FluxesBroadband object
+        real2d sw_flux_up ("sw_flux_up" ,ncol,nlay+1);
+        real2d sw_flux_dn ("sw_flux_dn" ,ncol,nlay+1);
+        real2d sw_flux_dn_dir("sw_flux_dn_dir",ncol,nlay+1);
+        real2d lw_flux_up ("lw_flux_up" ,ncol,nlay+1);
+        real2d lw_flux_dn ("lw_flux_dn" ,ncol,nlay+1);
+
+        // Run RRTMGP standalone codes and compare with AD run
+        // Do something interesting here...
+        // NOTE: these will get replaced with AD stuff that handles these
+        rrtmgp::rrtmgp_main(
+            p_lay, t_lay, p_lev, t_lev, gas_concs, col_dry,
+            sfc_alb_dir, sfc_alb_dif, mu0,
+            lwp, iwp, rel, rei,
+            sw_flux_up, sw_flux_dn, sw_flux_dn_dir,
+            lw_flux_up, lw_flux_dn
+        );
+
+        // Check values
+        REQUIRE(rrtmgpTest::all_equals(sw_flux_up_ref    , sw_flux_up    ));
+        REQUIRE(rrtmgpTest::all_equals(sw_flux_dn_ref    , sw_flux_dn    ));
+        REQUIRE(rrtmgpTest::all_equals(sw_flux_dn_dir_ref, sw_flux_dn_dir));
+        REQUIRE(rrtmgpTest::all_equals(lw_flux_up_ref    , lw_flux_up    ));
+        REQUIRE(rrtmgpTest::all_equals(lw_flux_dn_ref    , lw_flux_dn    ));
+
+        // Clean up from test; this is probably not necessary, these things
+        // should be deallocated when they fall out of scope, but we should be
+        // good citizens and clean up our mess.
+        p_lay.deallocate();
+        t_lay.deallocate();
+        p_lev.deallocate();
+        t_lev.deallocate();
+        col_dry.deallocate();
+        sfc_alb_dir.deallocate();
+        sfc_alb_dif.deallocate();
+        mu0.deallocate();
+        lwp.deallocate();
+        iwp.deallocate();
+        rel.deallocate();
+        rei.deallocate();
+        sw_flux_up_ref.deallocate();
+        sw_flux_dn_ref.deallocate();
+        sw_flux_dn_dir_ref.deallocate();
+        lw_flux_up_ref.deallocate();
+        lw_flux_dn_ref.deallocate();
+        sw_flux_up.deallocate();
+        sw_flux_dn.deallocate();
+        sw_flux_dn_dir.deallocate();
+        lw_flux_up.deallocate();
+        lw_flux_dn.deallocate();
+
+        gas_concs.reset();
+        rrtmgp::rrtmgp_finalize();
+        yakl::finalize();
+
+        // Make sure we exit cleanly
+        REQUIRE(true);
+    }
+        
+    /* 
+     * Run standalone test through SCREAM driver this time
+     */
+    TEST_CASE("rrtmgp_scream_stand_alone", "") {
         using namespace scream;
         using namespace scream::control;
 
-        /* 
-         * Setup driver stuff
-         */
+        // Initialize yakl
+        yakl::init();
```

Comment:
Every TEST_CASE causes this file to be executed multiple times _within the same run_. Is YAKL ok with being inited/finalized multiple times during the same run?

Kokkos, for instance, is not happy if you try to re-initialize it after it's been finalized.